.gitlab-ci.yml

```diff
@@ -0,0 +1,74 @@
+stages:
+  - build
+  - test
+  - report
+
+before_script:
+  - sh -c 'echo "deb [arch=amd64,arm64] http://repo.ros2.org/ubuntu/main `lsb_release -cs` main" > /etc/apt/sources.list.d/ros2-latest.list'
```

Comment:
@pbaughman I'm curious. What is the point of running the command explicitly with `sh` here?

Reply:
It's probably unnecessary.  I copy/pasted [from here](https://index.ros.org/doc/ros2/Installation/Linux-Install-Debians/)

> sudo sh -c 'echo "deb [arch=amd64,arm64] http://packages.ros.org/ros2/ubuntu `lsb_release -cs` main" > /etc/apt/sources.list.d/ros2-latest.list'

The sh -c was probably just to make the whole thing happen under sudo.  The docker image didn't like sudo (i'm already root in the docker image) so I got rid of it, but not the sh -c part.  I will remove it

Reply:
@pbaughman Oh since you mentioned `sudo` I understand why they use `sh -c`. 

If you `sudo echo xxxxx > /etc/qqqqq`, it'll fail as `sodu` only works for the part before `>`, the output redirection. This is a bash feature.